woocommerce/google-listings-and-ads
# Update Google Ads API to V14 and Content API packages

### Changes proposed in this Pull Request:



Project thread: pcTzPl-1K5-p2

This PR update:
 - The Google Ads library from 19.1.0 to 19.2.0 to support the Ads API V14.
 - The latest Google Content API version is still 2.15 so we only update its packages.
 - Change all imports from V13 > V14
 - Cleanup older versions of the Ads library from V11 to V13
 - Removed `vendor/guzzlehttp/promises/src/functions_include.php` from the autoload as `guzzlehttp/promises` no longer has an autoload for a `functions_include.php` file.
 - Removed `'GuzzleHttp\Promise\promise_for` from `bin/prefix-vendor-namespace.php` as it is not used anymore.
 
 ### Updated packages
 
 ```
 composer update googleads/google-ads-php --with-all-dependencies

  - Upgrading firebase/php-jwt (v6.5.0 => v6.8.1)
  - Upgrading google/common-protos (v4.1.0 => v4.2.0)
  - Upgrading google/gax (v1.20.2 => v1.22.1)
  - Upgrading google/protobuf (v3.23.2 => v3.24.0)
  - Upgrading googleads/google-ads-php (v19.1.0 => v19.2.0)
  - Upgrading guzzlehttp/promises (1.5.3 => 2.0.1)
  - Upgrading guzzlehttp/psr7 (2.5.0 => 2.6.0)

 ```
 
 ```
 composer update google/apiclient --with-all-dependencies

  - Upgrading google/apiclient-services (v0.302.0 => v0.312.0)
  - Upgrading phpseclib/phpseclib (3.0.19 => 3.0.21)

 ```

The most significant change is `guzzlehttp/promises (1.5.3 => 2.0.1)` which adds [PHP-type hints and removed support for the deprecated functions](https://github.com/guzzle/promises/compare/1.5.3...2.0.1) in `vendor/guzzlehttp/promises/src/functions_include.php`. Since it's already prefixed as part of guzzlehttp, we shouldn't run into any issues with other extensions.



### Detailed test instructions:


1. Checkout this PR, remove the vendors folder and run `composer install`
2. Run the unit tests locally to confirm they all pass without errors or warnings.
3. Test the plugin and the Connection Test page and confirm we can send requests without any fatal errors (need to connect all accounts).
4. Disconnect all accounts and do the onboarding.
5. Create an Ad Campaign.
6. Edit a campaign.
7. View the list of campaigns on the Dashboard page including some report data to confirm that there are no errors when using the updated library. In case you don't have live data for reporting, you can use these instructions: p1642505834083200-slack-C01E9LB4X61, which includes reporting data for the year 2021.
8. Send both regular requests to Merchant Center and Ads as well as proxied requests on the Connection Test page.

### Additional details:

Previously, we encountered conflicts with the following plugins::
- [flexible shipping ups](https://wordpress.org/plugins/flexible-shipping-ups/) https://github.com/woocommerce/google-listings-and-ads/pull/1977
- [backwpup](https://wordpress.org/plugins/backwpup/) https://github.com/woocommerce/google-listings-and-ads/pull/1980

I've tested these plugins, and everything seems to be working OK.


### Changelog entry

- Update - Google Ads library to API V14. (package v19.2.0)
- Update - Google API Client Services package to v0.312.
